docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+There are more insights regarding TensorFlow specifics, but it's too much to cover here.
+
+Now that our data can be retrieved and loaded, we will need some way to prepare it for our model and training.
+
+### 1.3.2 `PREVIEW` mode
+
+First, we put in our new `PREVIEW` mode code and functionality.
+To better understand TensorFlow usage, please recheck the companion piece for a more detailed explanation of each line.
+_(NOTE: we will assume the reader is doing this or understands `keras` and `tensorflow` moving forward)_
+****
+
+#### Function: `send_preview_data`
+
+```python
+def send_preview_data(self):
+  raw_train_ds = tf.data.Dataset.load(f"{self.data_url}/train")
+  conf = self.provider.full_config
+  # format for table values
+  samples = []
+  text_batch, label_batch = [n for n in raw_train_ds.take(1)][0]
+  samples = {
+    'textBatch': text_batch.numpy()[:2],
+    'labelBatch': label_batch.numpy()[:2]
+  }
+  conf['Configuration']['datasetInfo']['rawTrainSample'] = samples
+
+  translation_val = conf['Configuration']['textVectorizationConfig']['translationRequest']['token']
+  if translation_val:
+    conf['Configuration']['textVectorizationConfig']['tokenChips'] = self.get_token_translation(translation_val)
+    self.info(f"Received value for translation: {str(translation_val)}")
+
+    self.provider.save_full_config(conf)
+```
+
+In brief, we load our `tf.data.Dataset` as defined previously in our `DATA` mode. Then, make some sample formats to send to our UI, or "frontend".
+Note it also calls two(2) other utility functions called...
+
+#### Function: `self.get_token_translation`
+
+```python
+
+# NOTE TO READER: we will not have a model until the next "MODE", but mentally note the code explanation here
+
+def get_token_translation(self, translationVal):
+  model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})
+
+  try:
+    vect_layer = model.layers[0]
+    translation_tokens = [token.strip(" ") for token in translationVal.split(",")]
+    vect_vocab = vect_layer.get_vocabulary()
+    translations = [{"token": int(token), "translation": vect_vocab[int(token)]} for token in translation_tokens]
+  
+    self.info(str(translations))
+    return translations
+  except Exception as e:
+    logger.error(f"get_token_translation: Error occured \n {repr(e)}")
+    self.provider.io.error("An error occured during token translation :( . Please see PythonSdk log for details")
+```
+
+`model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})`
+
+Here we load a model that(soon) will contain a `TextVectorization` TensorFlow layer at `model.layers[0]` where `translational` is a comma-separated value of integer token values to be "translated" to their original string representations as mapped by said `model.layers[0]` layer.
```

Comment:
TW review:

Here we load a model that will soon contain a `TextVectorization` TensorFlow layer at `model.layers[0]` where `translational` is a comma-separated value of integer token values to be "translated" to their original string representations, as mapped by said `model.layers[0]` layer.